### DISCUSSION TITLE
How to reboot mirthconnect mcserver automatically

### DISCUSSION THREAD

**Discussion by sal4455** [nextgenhealthcare/connect#5154] [General]
Hi all ,
Hope you are well , 
I just want to know , can mirthconnect server(mcserver) be rebooted incase of any failure ?

**Comment by pacmano1** [comment]
It can be rebooted.  Your question does not make sense to me however.  "Reboot" is not the same thing as "restart".   What do you mean by automatically? If you're looking for Mirth to be self healing and identify on its own that it has failed, that would need to occur via normal operating system capabilities rather than Mirth itself. I believe in your previous post you're looking to start the server with python rather than use the normal Mirth startup scripts (in linux) or via Windows Services.  In Linux folks sometimes use "monit" to restart failed services.  In Windows service manager there is an option to restart a service on failure.  This is how Mirth is intended to be used - i.e. as a server process running all the time.  Mirth is much like other server processes that run all the time. Again, I think you may not fully understand how and when to use Mirth.    Mirth is very reliable.

**Reply by sal4455** [reply]
Hey @pacmano1 , I understood what you're trying to say. 
Earlier my use case was different, now we are using mirthconnect as a service means once started it will run all the time. 
As I am using mirth in linux, so my question was- is there any bash script that we can leverage to reboot it if mirth fails .

"If you're looking for Mirth to be self healing and identify on its own that it has failed, that would need to occur via normal operating system capabilities rather than Mirth itself"--  yes I am looking for it .

**Reply by pacmano1** [reply]
I already mentioned the tool "monit".  It is free.   Read https://mmonit.com/monit/.    

Also, for production purposes, make sure mirth is using a real database for its own data store rather that "Derby" which is its default database.

However, mirth is VERY stable,  many users with very high loads don't need to use a tool like monit to check if it is running and restart it.   If Mirth fails it is usually due to some outside factor or channel design problem, both of which will typically persist after the restart if you don't fix them.
